constraint-solver/src/solver.rs

```diff
@@ -203,13 +229,58 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
             false
         }
     }
+
+    /// Given a set of variable assignments, checks if they satisfy all the constraints.
+    /// Note that this might return false positives, because it does not propagate any values.
+    fn check_assignments(&self, assignments: &BTreeMap<V, T>) -> bool {
```

Comment:
Maybe `is_assignment_conflicting` - to indicate the meaning of the return value (of course it needs to be inverted)?